src/main/java/AlgorithmicGroupTheory/MORGEN.java

```diff
@@ -1126,7 +1136,7 @@ public static boolean blockTest(int r, int[][] A) {
 	}
 	
 	public static void clearFormers(boolean check, int y) {
-		 if(check==false) {
+		 if(!check) {
 			 for(int i=y;i<formerPermutations.size();i++) {
 				 formerPermutations.get(i).removeAll(formerPermutations.get(i));

```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [DMI: Don’t use removeAll to clear a collection (DMI_USING_REMOVEALL_TO_CLEAR_COLLECTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dmi-don-t-use-removeall-to-clear-a-collection-dmi-using-removeall-to-clear-collection).
If you want to remove all elements from a collection c, use c.clear, not c.removeAll(c). Calling c.removeAll(c) to clear a collection is less clear, susceptible to errors from typos, less efficient and for some collections, might throw a ConcurrentModificationException.

Reply:
Good catch. I will try to fix it.

Reply:
I added `clear()` call. Thank you for suggestion.